model/parts/utils/ethereum_spec.py

```diff
@@ -13,6 +13,38 @@
 
 # Beacon state accessors
 
+def get_active_validator_indices(state: StateVariables) -> Gwei:
+    """
+    See https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#get_active_validator_indices
+
+    ```python
+    def get_active_validator_indices(state: BeaconState, epoch: Epoch) -> Sequence[ValidatorIndex]:
+        '''
+        Return the sequence of active validator indices at ``epoch``.
+        '''
+        return [ValidatorIndex(i) for i, v in enumerate(state.validators) if is_active_validator(v, epoch)]
+    ```
+    """
+    return state["number_of_active_validators"]
+
+
+def get_awake_validator_indices(params: Parameters, state: StateVariables) -> Gwei:
+    """
+    Simplified Active Validator Cap and Rotation Proposal
+
+    See https://ethresear.ch/t/simplified-active-validator-cap-and-rotation-proposal
+    """
+    # Parameters
+    MAX_VALIDATOR_COUNT = params['MAX_VALIDATOR_COUNT']
+
+    # State Variables
+    active_validators = get_active_validator_indices(state)
+
+    # Get awake validators as subset of active validators
+    awake_validators = min(MAX_VALIDATOR_COUNT, active_validators)
```

Comment:
Using `min()` with a `None` type will break the code, I suggest the following (it works both for `None` and `float('inf')`):

```suggestion
    awake_validators = min(x for x in [MAX_VALIDATOR_COUNT, active_validators] if x is not None)
```